### PR TITLE
Refactor wallet BTC balance parameters

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Closes: #220 
Closes: #180 
Depends on: #232

So far, there were two parameters responsible for controlling the wallet creation and closure:
- `walletMinBtcBalance`
- `walletMaxBtcBalance`

The purpose of the latter parameter was clear since it was used for closure exclusively but the former was ambiguous as it was referring to both creation and closure. In order to break that ambiguity, we refactor those parameters to the following form:
- `walletCreationMinBtcBalance`
- `walletCreationMaxBtcBalance`
- `walletClosureMinBtcBalance`

Now, each parameter clearly refers to either wallet creation or closing.

### CODE REVIEW

File: solidity/contracts/bridge/BridgeState.sol [hunk 1]
```diff
@@ -175,11 +175,14 @@ library BridgeState {
...
-        uint64 walletMinBtcBalance;
+        // wallet creation.
+        uint64 walletCreationMinBtcBalance;
         // The maximum BTC threshold in satoshi that is used to decide about
         // wallet creation.
-        uint64 walletMaxBtcBalance;
+        uint64 walletCreationMaxBtcBalance;
+        // The minimum BTC threshold in satoshi that is used to decide about
+        // wallet closing.
+        uint64 walletClosureMinBtcBalance;
```
Comment (tomaszslabon) [inline solidity/contracts/bridge/BridgeState.sol:185]: I would give these three parameters a more precise description as how they decide about wallet creation or closure, similar to how they are described in
[RFC](https://github.com/keep-network/tbtc-v2/blob/main/docs/rfc/rfc-1.adoc#wallet-lifecycle):
```
wallet_min_closure_btc: The smallest amount of btc a wallet can hold before we attempt to close the wallet and transfer the funds to a randomly selected wallet.

```

Comment (tomaszslabon) [inline solidity/contracts/bridge/BridgeState.sol:185]: I would give these three parameters a more precise description as how they decide about wallet creation or closure, similar to how they are described in
[RFC](https://github.com/keep-network/tbtc-v2/blob/main/docs/rfc/rfc-1.adoc#wallet-lifecycle):
```
wallet_min_closure_btc: The smallest amount of btc a wallet can hold before we attempt to close the wallet and transfer the funds to a randomly selected wallet.

```

Comment (lukasz-zimnoch) [reply solidity/contracts/bridge/BridgeState.sol]: Good idea! Here it is: https://github.com/keep-network/tbtc-v2/pull/233/commits/d4e6c538a0b24575512250fc5e3a7bec2b8d6971

File: solidity/contracts/bridge/BridgeState.sol [hunk 2]
```diff
@@ -175,11 +175,20 @@ library BridgeState {
...
+        // wallet if the active wallet is old enough and their amount of BTC
+        // is greater than or equal this threshold.
+        uint64 walletCreationMinBtcBalance;
         // The maximum BTC threshold in satoshi that is used to decide about
-        // wallet creation.
-        uint64 walletMaxBtcBalance;
+        // wallet creation. Specifically, we allow for the creation of a new
+        // wallet if the active wallet's amount of BTC is greater than or equal
+        // this threshold, regardless of the active wallet's age.
+        uint64 walletCreationMaxBtcBalance;
```
Comment (pdyraga) [inline solidity/contracts/bridge/BridgeState.sol:186]: RFC 1 tells about `wallet_min_creation_btc` and `wallet_min_closure_btc` but nothing about `wallet_max_creation_btc`. As far as I recall, it was removed some time ago. We need to decide if it's needed or not and aligned the RFC and implementation cc @beaushinkle

Comment (lukasz-zimnoch) [reply solidity/contracts/bridge/BridgeState.sol:186]: IIRC it was not mentioned in the RFC but decided out of band to have this parameter in order to allow creating new wallets regardless of the active wallet age once the active wallet has too much BTC.

**Comment by pdyraga** [comment]
👀

File: solidity/contracts/bridge/Bridge.sol [hunk 3]
```diff
@@ -235,8 +236,9 @@ contract Bridge is Governable, EcdsaWalletOwner {
         self.fraudChallengeDefeatTimeout = 7 days;
         self.fraudChallengeDepositAmount = 2 ether;
         self.walletCreationPeriod = 1 weeks;
-        self.walletMinBtcBalance = 1e8; // 1 BTC
-        self.walletMaxBtcBalance = 10e8; // 10 BTC
+        self.walletCreationMinBtcBalance = 1e8; // 1 BTC
+        self.walletCreationMaxBtcBalance = 10e8; // 10 BTC
```
[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply solidity/contracts/bridge/Bridge.sol]: Sure, see https://github.com/keep-network/tbtc-v2/pull/233/commits/467401a0c17a45b82e2ab2f1451d03b35b635cc8

File: solidity/contracts/bridge/BridgeState.sol [hunk 4]
```diff
@@ -449,19 +461,24 @@ library BridgeState {
...
+        uint64 _walletCreationMaxBtcBalance,
+        uint64 _walletClosureMinBtcBalance,
         uint32 _walletMaxAge,
         uint64 _walletMaxBtcTransfer,
         uint32 _walletClosingPeriod
     ) internal {
         require(
-            _walletMinBtcBalance > 0,
-            "Wallet minimum BTC balance must be greater than zero"
+            _walletCreationMinBtcBalance > 0,
```
[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply solidity/contracts/bridge/BridgeState.sol]: Makes sense, see https://github.com/keep-network/tbtc-v2/pull/233/commits/6956770281f7ba4c74a854d07f88a446f2a98dc9

**Review by pdyraga — APPROVED** [review]

**Pull request merged**